### ISSUE TITLE
Deepcopy for networks

### ISSUE THREAD

**Issue by z3ustb** [eth-brownie/brownie#1428]
https://github.com/eth-brownie/brownie/blob/9a3bd8afe3e3c173f0847e8ef10769bb923bef46/brownie/_config.py#L76

As the dict.copy() is a shallow copy function for dictionary, should the deepcopy be used here?

**Comment by WyseNynja** [comment]
Yes, it should. https://github.com/eth-brownie/brownie/pull/1466
